File: C#/Command/FileHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Command
{
    class FileHandler
    {
        private string fileName;
        private DateTime date;
        private bool alreadySaved;

        public FileHandler()
        {
            this.fileName = "";
            date = DateTime.Now;
            alreadySaved = false;
        }

        public bool SaveFile(string fileName)
        {
            if (alreadySaved) return false;
            Console.WriteLine($"\nSaved file: {fileName} on + {date}");
            alreadySaved = true;
            return true;
        }

        public void ReadFile(string fileName)
        {
            if (alreadySaved == false)
            {
                Console.WriteLine($"\nCan not read new file before saving current file\n");
            }
            else
            {
                Console.WriteLine($"\nReading file: {fileName}\n");
            }
        }
    }
}

File: python/Command/FileHandler.py
import string
import datetime

class FileHandler:
    def __init__(self) -> None:
        self.fileName = ""
        self.date = datetime.date.today()
        self.alreadySaved = False

    def SaveFile(self, fileName: string) -> None:
        if not (self.alreadySaved):
            print(f"Saved file: {fileName} on {str(self.date)}")
            self.alreadySaved = True

    def ReadFile(self, fileName: string) -> None:
        if (self.alreadySaved == False):
            print("\nCan not read new file before saving current file\n")
        else:
            print(f"\nReading file: {fileName}\n")